### PR TITLE
Nickname handling in create pane

### LINKED ISSUE #127: Account Name in Transfer Transaction
### Description

When creating a transfer transaction, if an account nickname contains a dash (-) in the name you cannot add the account to the From or To Accounts.

### Steps to reproduce

1. Add Account Info containing Nickname with a dash (e.g. Account-Nickname)
2. Click on Create
3. Select Transfer from the Transaction Type
4. Under From and To accounts enter the account number and HBAR amount
5. Try to click Accept to add to list

### Additional context

_No response_

### Hedera network

other

### Version

v0.3.0

### Operating system

macOS

### PR DESCRIPTION
**Pull request by DSMatusevich**
**Description**:
Refactored the parsing of account strings to allow the use of special characters in nicknames.

**Related issue(s)**:

Closes #127 

**Checklist**

- [x] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

File: tools-core/src/main/java/com/hedera/hashgraph/client/core/json/Identifier.java [hunk 1]
```diff
@@ -158,13 +162,27 @@ private static void handleShardOrRealmNumber(JsonObject jsonObject, String field
...
-		var idC = id.contains("(") ? id.substring(id.indexOf("(") + 1, id.indexOf(")")) : id;
+		String idC = "";
+
+		var pattern1 = Pattern.compile(FULL_ACCOUNT_CHECKSUM_REGEX);
+		var pattern2 = Pattern.compile(FULL_ACCOUNT_REGEX);
+		var pattern3 = Pattern.compile(NUMBER_REGEX);
+
+		var matcher1 = pattern1.matcher(id);
+		var matcher2 = pattern2.matcher(id);
+		var matcher3 = pattern3.matcher(id);
```
Comment (qnswirlds) [inline tools-core/src/main/java/com/hedera/hashgraph/client/core/json/Identifier.java:177]: nit: those could also be `final`

File: tools-core/src/main/java/com/hedera/hashgraph/client/core/utils/CommonMethods.java [hunk 2]
```diff
@@ -590,5 +594,33 @@ public static String splitString(String digest) {
...
+		var matcherAccount = patternAccount.matcher(value);
+		if (matcherAccount.find()) {
+			var identifier = Identifier.parse(matcherAccount.group(0));
+			return identifier.toReadableStringAndChecksum();
+		}
+		var patternDecimal = Pattern.compile(NUMBER_REGEX);
+		var matcherDecimal = patternDecimal.matcher(value);
+		if (matcherDecimal.find()) {
+			var identifier = Identifier.parse(matcherDecimal.group(0));
+			return identifier.toReadableStringAndChecksum();
```
Comment (qnswirlds) [inline tools-core/src/main/java/com/hedera/hashgraph/client/core/utils/CommonMethods.java:621]: Same here, add `final` where appropriate

**Review by qnswirlds — DISMISSED** [review]
Changes LGTM. Should add `final` where appropriate to all the files that were touched.

**Review by qnswirlds — APPROVED** [review]
`final` were missing in other files as well, not only the 2 files I pointed out. But I don't mind re-approving this PR.

**Comment by DSMatusevich** [comment]
> `final` were missing in other files as well, not only the 2 files I pointed out. But I don't mind re-approving this PR.

I'm trying to solve those problems as I see them or are pointed out to me. At some point I will do a review of the whole code and add all the missing `final`

**Pull request merged**